flow/layers/container_layer.cc

```diff
@@ -72,12 +72,16 @@ void ContainerLayer::PrerollChildren(PrerollContext* context,
 }
 
 void ContainerLayer::PaintChildren(PaintContext& context) const {
-  FML_DCHECK(needs_painting());
+  // needs_painting(context) is only valid for the PaintContext that
+  // is initially handed to a layer's Paint() method. By the time the
+  // layer calls PaintChildren(), though, it may have modified the
+  // PaintContext so we can no longer double-check the culling.
+  // FML_DCHECK(needs_painting(context));
```

Comment:
We prefer removing the code instead of just commenting it out. You can still leave the comment here with something like `Don't FML_DCHECK(needs_painting(context)) here! needs_painting(context) is only valid ...`.

Reply:
new comment pushed